### ISSUE TITLE
Sounds Shaders (or Toner?)

### ISSUE THREAD

**Issue by Kangz** [Unvanquished/Unvanquished#357]
For now sound are just "samples" with no information attached: the gamelogic and the map need to handle special effects by hand.

Sound shaders would be sets of parameters associated to one or more sound file that could be registered transparently ("foo" could be foo.ogg foo.wav or foo.soundshader) Possible effects would be randomizing the samples, having a loop, changing the attenuation because of distance, ... See sound shaders in D3 for some ideas

It's been decided that sound shaders will be written in YAML so that we don't have to use our own parser (slowly transitioning the rest of the engine).
- Mentor: Kangz
- Knowledge: Audio, Parser, Resource model
- Difficulty: 6

[Mentored projects on the Unv wiki](http://unvnet.net/wiki/index.php/Mentored_Coding_Project)

**Comment by illwieckz** [comment]
the very basic of sound shader would have to tweak the amplification, to not have to push a new file into repository just because the volume was too loud or not enough…

**Comment by Viech** [comment]
> the very basic of sound shader would have to tweak the amplification, to not have to push a new file into repository just because the volume was too loud or not enough…

Agreed, just stumbled upon this problem and the minimum functionality that would make this useful is just volume adjustment.
